boards/posix/simple_process/hw_models_top.c

```diff
@@ -0,0 +1,137 @@
+/*
```

Comment:
It's unclear to me why this can't be in drivers/ instead of in boards/. Maybe I'm missing something, but won't this be common to all `boards/posix/*` boards?

Reply:
Nope, this is not common to all boards. This is just one way of doing it, even if a generic way.
The thing here is that, AFAIK, HW models do not fit anywhere in the current zephyr tree. So I just assumed the logical place for them was hidden inside the board folder, to keep the thing contained.